DynamicTablesPkg/Include/ArchCommonNameSpaceObjects.h

```diff
@@ -728,6 +731,54 @@ typedef struct CmArchCommonObjSpmiInterruptDeviceInfo {
   /** Uid of the device */
   UINT32    DeviceId;
 } CM_ARCH_COMMON_SPMI_INTERRUPT_DEVICE_INFO;
+
+/** A structure that describes the Cst information.
+
+  Processor power state (C-state) is described in DSDT/SSDT and associated
+  to cpus/clusters in the cpu topology.
+
+  Unsupported Optional registers should be encoded with NULL resource
+  Register {(SystemMemory, 0, 0, 0, 0)}
+
+  For values that support Integer or Buffer, integer will be used
+  if buffer is NULL resource.
+  If resource is not NULL then Integer must be 0
+
+  Cf. ACPI 6.4, s8.4.2.1 _CST (C states)
+
+  ID: EArchCommonObjCstInfo
+*/
+typedef AML_CST_INFO CM_ARCH_COMMON_CST_INFO;
+
+/** A structure that describes the C-State Dependency (CSD) Info.
+
+    Cf. ACPI 6.4, s8.4.2.2 _CSD (C-State Dependency).
+
+    ID: EArchCommonObjCsdInfo
+*/
+typedef AML_CSD_INFO CM_ARCH_COMMON_CSD_INFO;
+
+/** A structure that describes the P-State Info.
+    This includes _PCT, _PSS and method.
+
+    Cf. ACPI 6.4, s8.4.6. Processor Performance Control
+
+    ID: EArchCommonObjPstateInfo
+*/
+typedef struct CmArchCommonPstateInfo {
+  /// Processor Control Structure (PCT)
+  AML_PCT_INFO    Pct;
+
+  /// Processor Status Structure (PSS)
+  AML_PSS_INFO    *Pss;

```

Comment:
Similar comment as for:
```AML_CST_PKG_INFO    *CStates;```
You will have to use `CM_ARCH_COMMON_OBJ_REF` unfortunately